src/js/components/sku-list.js

```diff
@@ -48,21 +49,56 @@ class SkuList extends LitElement {
     this.locale = 'en-US';
     this.type = 'sku';
   }
+
+  /**
+   *
+   * @param {PurchaseDetails} basicSubPurchase
+   * @param {PurchaseDetails} premiumSubPurchase
+   * @param {string} itemId
+   * @return {string}
+   */
+  skuType(basicSubPurchase, premiumSubPurchase, itemId) {
+    if (itemId === BASIC_SUB && premiumSubPurchase) {
+      return 'downgrade';
+    } else if (itemId === PREMIUM_SUB && basicSubPurchase) {
+      return 'upgrade';
+    } else {
+      return this.type;
+    }
+  }
+
   /**
    * @return {TemplateResult}
    */
   render() {
+    const basicSubPurchase = this.purchases.find((purchase) => purchase.itemId === BASIC_SUB);
+    const premiumSubPurchase = this.purchases.find((purchase) => purchase.itemId === PREMIUM_SUB);
     return html`${this.skus.map((sku) => {
       // Find if there's a purchase with the same itemId as the SKU
       const purchase = this.purchases.find((purchase) => purchase.itemId === sku.itemId);
+      const skuType = this.skuType(basicSubPurchase, premiumSubPurchase, sku.itemId);
       return html` <sku-holder
-        .type="${this.type}"
+        .type="${skuType}"
         .details=${sku}
         price="${this.service.getSkuPrice(sku, this.locale)}"
         .purchase=${purchase
           ? null
           : async function () {
-              const { response, valid } = await this.service.purchase(sku.itemId);
+              let purchaseMade;
+              switch (skuType) {
+                case 'downgrade':
+                  // downgrade from premium to basic subscription
+                  purchaseMade = await this.service.purchase(sku.itemId, premiumSubPurchase);
```

Comment:
is premiumSubPurchase here the premium subscription token?

Reply:
It's the whole `PurchaseDetails`. In play-billing.js lines 170-171, we need to pass in the purchase item ID and purchase token there. So, I pass in the whole `PurchaseDetails` to the `purchase` method.